src/deadline/nuke_submitter/ui/components/scene_settings_tab.py

```diff
@@ -48,31 +51,140 @@ def _build_ui(self):
             self.write_node_box.addItem(write_node.fullName(), write_node.fullName())
 
         lyt.addWidget(QLabel("Write Nodes"), 0, 0)
-        lyt.addWidget(self.write_node_box, 0, 1)
+        lyt.addWidget(self.write_node_box, 0, 1, 1, -1)
 
         self.views_box = QComboBox(self)
         self.views_box.addItem("All Views", "")
         for view in sorted(nuke.views()):
             self.views_box.addItem(view, view)
         lyt.addWidget(QLabel("Views"), 1, 0)
-        lyt.addWidget(self.views_box, 1, 1)
+        lyt.addWidget(self.views_box, 1, 1, 1, -1)
 
         self.frame_override_chck = QCheckBox("Override Frame Range", self)
         self.frame_override_txt = QLineEdit(self)
         lyt.addWidget(self.frame_override_chck, 2, 0)
-        lyt.addWidget(self.frame_override_txt, 2, 1)
+        lyt.addWidget(self.frame_override_txt, 2, 1, 1, -1)
         self.frame_override_chck.stateChanged.connect(self.activate_frame_override_changed)
 
         self.proxy_mode_check = QCheckBox("Use Proxy Mode", self)
         lyt.addWidget(self.proxy_mode_check, 3, 0)
 
+        self.timeout_checkbox = QCheckBox("Use Timeouts", self)
+        self.timeout_checkbox.setChecked(True)
+        self.timeout_checkbox.clicked.connect(self.activate_timeout_changed)
+        self.timeout_checkbox.setToolTip(
+            "Set a maximum duration for actions from this job. See AWS Deadline Cloud documentation to learn more"
+        )
+        lyt.addWidget(self.timeout_checkbox, 4, 0)
+        self.timeouts_subtext = QLabel("Set a maximum duration for actions from this job")
+        self.timeouts_subtext.setStyleSheet("font-style: italic")
+        lyt.addWidget(self.timeouts_subtext, 4, 1, 1, -1)
+
+        self.timeouts_box = QGroupBox()
+        timeouts_lyt = QGridLayout(self.timeouts_box)
+        lyt.addWidget(self.timeouts_box, 5, 0, 1, -1)
+
+        def create_timeout_row(label, tooltip, row):
+            qlabel = QLabel(label)
+            qlabel.setToolTip(tooltip)
+            timeouts_lyt.addWidget(qlabel, row, 0)
+
+            days_box = QSpinBox(self, minimum=0, maximum=365)
+            days_box.setSuffix(" days")
```

Comment:
Nit: this text should also support a value of `1`. E.g. "1 day" instead of "1 days"

Reply:
This would apply to hours and minutes as well. Quick change is parenthesis: `day(s)`, better fix is update suffix on value change?

Maybe just the quick change for now?

Reply:
> better fix is update suffix on value change?

We already had a callback setup for the spin box value change to determine if the sum was 0 valued and mark the label as red. I've added additional logic there to also check if any of the suffixes should be singular